src/main/java/net/mcreator/blockly/BlocklyToCode.java

```diff
@@ -317,4 +324,14 @@ public Collection<String> getUsedBlocks() {
 		return Collections.unmodifiableSet(usedBlocks);
 	}
 
+	private record ExtraTemplatesLinker(BlocklyToCode master) implements TemplateDirectiveModel {

```

Comment:
This already exists, to not duplicate this, make this external class instead, for example

Reply:
You can even put it into this same package and then make addTemplate method package private